crates/paymaster-execution/src/diagnostics/extractors/avnu.rs

```diff
@@ -0,0 +1,1135 @@
+//! AVNU Exchange contract metadata extractor.
+//!
+//! Extracts diagnostic information from failed AVNU swap transactions,
+//! particularly focusing on slippage errors and swap parameters.
+
+use async_trait::async_trait;
+use starknet::core::types::{Call, Felt};
+use std::collections::HashMap;
+
+use crate::diagnostics::{CallDiagnostic, CallMetadataExtractor, DiagnosticContext, DiagnosticMetric, DiagnosticValue, ErrorCategory};
+use crate::tokens::TokenService;
+
+/// AVNU Exchange contract address on Starknet mainnet.
+pub const AVNU_EXCHANGE_ADDRESS: Felt = Felt::from_hex_unchecked("0x04270219d365d6b017231b52e92b3fb5d7c8378b05e9abc97724537a80e93b0f");
+
+/// Converts a Felt to u128, returning 0 if the value doesn't fit.
+fn felt_to_u128(felt: Felt) -> u128 {
```

Comment:
This is a utility method, can be moved in avnu-starknet, maybe it even already exists ?